include/gmsh.tcc

```diff
@@ -109,42 +110,142 @@ void GMSH::read_elements(const std::string& filename) {
         istream >> elementry;
 
         //! \brief Check element type
-        //! \details If element type not == to Tdim, skip element
-        if (elementtype != ndimension) {
+        //! \details If element type not == to specified Tvertices, skip
+        //! element
+        if (elementtype != Tvertices) {
           istream >> line;
         } else {
           istream >> elementarray.at(0) >> elementarray.at(1) >>
               elementarray.at(2) >> elementarray.at(3);
-          elements_.emplace_back(
-              new Point<ndimension>(elementid, elementarray));
+
+          Mesh<Tdim, Tvertices>::elements_.insert(
+              std::make_pair(elementid, elementarray));
         }
       }
     }
     infile.close();
   }
+  std::cout << "Number of Elements: " << Mesh<Tdim, Tvertices>::elements_.size()
+            << '\n';
+
+  //! Get the coordinates for each vertex of each element
+  GMSH::store_element_vertices();
+}
+
+//! \brief store each coordinates of an element
+//! \Param[in] Tdim = Dimension, Tvert = n vertices in element
+template <unsigned Tdim, unsigned Tvertices>
+void GMSH<Tdim, Tvertices>::store_element_vertices() {
+
+  const unsigned firstelement = Mesh<Tdim, Tvertices>::elements_.begin()->first;
+  const unsigned lastelement = Mesh<Tdim, Tvertices>::elements_.rbegin()->first;
+
+  typename std::map<unsigned, std::array<double, Tvertices>>::iterator
+      elementfind;
+  typename std::map<unsigned, std::array<double, Tdim>>::iterator verticesfind;
+
+  std::array<double, Tvertices> elementkeyvalues;
+  std::array<double, Tdim * Tvertices> verticesarray;
+
+  //! Iterate through element_
+  for (unsigned i = firstelement; i <= lastelement; ++i) {
+    elementfind = Mesh<Tdim, Tvertices>::elements_.find(i);
+    if (elementfind != Mesh<Tdim, Tvertices>::elements_.end()) {
+
+      //! In each element, iterate to get vertices id's of the element
+      for (unsigned j = 0; j < 4; ++j) {
```

Comment:
Can we not combine these two for loops? Shouldn't this be iterating to Tvertices?